### DISCUSSION TITLE
{{yaml_value:property_name}} variable

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#56] [Ideas - done & released]
`{{yaml:variablename}}`: This variable would give the value of a single YAML property. It should be something similar to how [Dataview plugin](https://github.com/blacksmithgu/obsidian-dataview) accesses YAML properties, maybe it could even utilize [Dataview's API](https://blacksmithgu.github.io/obsidian-dataview/plugin/develop-against-dataview/) to access the property - if it helps. I'm not familiar with Dataview's API, so I do not know what it offers.

Alternative names: `{{frontmatter:variablename}}`, `{{header:variablename}}`.

**Comment by FelipeRearden** [comment]
Great ideia @Taitava !!!!

From #57 : `{{yaml_content}}` --> We need to think if we are gonna include `---` from the yaml template or not 

![Screen Shot 2021-09-30 at 11 50 19](https://user-images.githubusercontent.com/85180696/135478729-f2999636-b94e-4ff1-8c06-56be78ebbc15.png)

**Reply by Taitava** [reply]
Yes, good point. `{{yaml_content}}` could have a parameter, e.g. `{{yaml_content:no-dashes}}` (or something).

**Comment by Taitava** [comment]
I decided to use a name `{{yaml_value}}` instead of just `{{yaml}}`. This allows to create logical new variants to be created later, e.g. `{{yaml_content}}`, `{{yaml_keys}}` and `{{yaml_values}}`, but those should be discussed in a new discussion.
